File: front end/rcf-soluciones-informaticas-angular/src/app/shared/services/product-services/product-attribute-value-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import ProductAttributeValue from '../../ProductAttributeValue';
import ProductAttributeValueRequest from '../../model/Products-Service/ProductAttributeValueRequest.model';

@Injectable({
  providedIn: 'root'
})
export class ProductAttributeValueService {
  private apiUrl = 'http://localhost:8080/products'; 
  private domain = 'product-attribute-values';
  constructor(private http: HttpClient) {
  }
  getProductAttributeValues(): Observable<ProductAttributeValue[]> {
    return this.http.get<ProductAttributeValue[]>(`${this.apiUrl}/${this.domain}`);
  }
  getProductAttributeValueById(id: number): Observable<ProductAttributeValue> {
    return this.http.get<ProductAttributeValue>(`${this.apiUrl}/${this.domain}/${id}`);
  }
  createProductAttributeValue(productAttributeValue: ProductAttributeValueRequest): Observable<ProductAttributeValue> {
    return this.http.post<ProductAttributeValue>(`${this.apiUrl}/${this.domain}`, productAttributeValue);
  }
  updateProductAttributeValue(id: number, productAttributeValue: ProductAttributeValueRequest): Observable<ProductAttributeValue> {
    return this.http.put<ProductAttributeValue>(`${this.apiUrl}/${this.domain}/${id}`, productAttributeValue);
  }
  deleteProductAttributeValue(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${this.domain}/${id}`);
  }
  getProductAttributeValuesByProductId(productId: number): Observable<ProductAttributeValue[]> {
    return this.http.get<ProductAttributeValue[]>(`${this.apiUrl}/${this.domain}/product/${productId}`);
  }
}

File: back end/rcf-soluciones-informaticas/products-service/src/main/java/com/rcf/productsservice/service/ProductAttributeValueService.java
package com.rcf.productsservice.service;

import com.rcf.productsservice.dto.ProductAttributeValueRequest;
import com.rcf.productsservice.dto.ProductAttributeValueResponse;

import java.util.List;

public interface ProductAttributeValueService {
    List<ProductAttributeValueResponse> getAllProductAttributeValues();
    ProductAttributeValueResponse getProductAttributeValueById(Long id);
    ProductAttributeValueResponse createProductAttributeValue(ProductAttributeValueRequest request);
    ProductAttributeValueResponse updateProductAttributeValue(Long id, ProductAttributeValueRequest request);
    void deleteProductAttributeValue(Long id);

    List<ProductAttributeValueResponse> getProductAttributeValuesByProductId(Long productId);
}
